docs/REAME-develop-add-XR-support.md

```diff
@@ -0,0 +1,277 @@
++       assert_raises(Cisco::UnsupportedError) do
++         neighbor.low_memory_exempt = true
++       end
++     else
+        check = [true, false, neighbor.default_low_memory_exempt]
+        check.each do |value|
+          neighbor.low_memory_exempt = value
+          assert_equal(value, neighbor.low_memory_exempt)
+        end
++     end
+      neighbor.destroy
+    end
+  end
+```
+
+#### <a name="interface">Interface differences</a>
+
+The interfaces available for use on XR will be different than Nexus, so any existing tests that refer to interfaces by name will likely fail on XR.  Here is the error from the update-source test:
+
+```bash
+  1) Error:
+TestRouterBgpNeighbor#test_update_source:
+Cisco::CliError: CliError: '(unknown, see error message)' rejected with message:
+'The command '(unknown, see error message)' was rejected with error:
+
+Unable to process cmd, ret-val: 3, cmd: router bgp 55 neighbor 1.1.1.1  update-source Ethernet1/1'
+```
+
+From the error, you can't tell which part of the command was rejected, so you might initially think that update-source is not supported on XR.  Typing the command directly into the XR CLI will give a better idea of what specifically is wrong.
+
+```bash
+RP/0/RP0/CPU0:agent-lab20-xr#conf
+Fri Jan 29 14:02:50.566 UTC
+RP/0/RP0/CPU0:agent-lab20-xr(config)#router bgp 55 neighbor 1.1.1.1  update-source Ethernet1/1
+                                                                                   ^
+% Invalid input detected at '^' marker.
+```
+
+As you can see in CLI output, the '^' marker is indicating that "Ethernet1/1" is invalid.  Instead of using hardcoded interface names, use the `interfaces[]` array (see [Development Best Practices: MT3](./README-develop-best-practices.md#mt3)).
+
+```ruby
+  def test_update_source
+    %w(default test_vrf).each do |vrf|
+      neighbor = create_neighbor(vrf)
+-     test_interfaces = ['loopback1', 'Ethernet1/1', 'ethernet1/1',
++     test_interfaces = ['loopback1', interfaces[0], interfaces[0].downcase,
+                         neighbor.default_update_source]
+      test_interfaces.each do |interface|
+        neighbor.update_source = interface
+        assert_equal(interface.downcase, neighbor.update_source)
+      end
+      neighbor.destroy
+    end
+  end
+```
+
+#### <a name="setup">Setup/teardown differences</a>
+
+Your minitest's setup method is run before each test, and the teardown method is run after each test.  These methods often contain calls to the `TestCase.config` method to execute commands to clear existing configuration on the device, and often, the commands will differ between platforms. Currently, the `config` method does not display the output from executed commands, even when debug is enabled, so any errors that occur will be masked.
+
+You should verify that any existing config commands are valid for XR (either by sight, or by executing them manually on the XR CLI).  The following changes were needed for `test_bgp_neighbor.rb`:
```

Comment:
General nit - we probably shouldn't be referring to 'XR' but should instead say 'IOS XR' throughout.